test/integration/stack-trace.js

```diff
@@ -0,0 +1,29 @@
+const assert = require('assert');
+const childProcess = require('child_process');
+
+(function main() {
+  if (process.env.CHILD !== 'stack-trace') {
+    const cp = childProcess.fork(__filename, [], {
+      stdio: 'pipe',
+      env: Object.assign({}, process.env, {
+        CHILD: 'stack-trace',
+      })
+    });
+    let stdoutBuffers = []
+    cp.stdout.on('data', data => stdoutBuffers.push(data));
+
+    cp.on('exit', (code) => {
+      const stdout = Buffer.concat(stdoutBuffers).toString('utf8')
+      if (code !== 0) {
+        process.stderr.write(stdout);
+        process.exit(1);
+        return;
+      }
+      assert(stdout.match(/evalmachine.<anonymous>/) == null);
+      assert(stdout.match(__filename + '.cache.js') != null);
+    });
+    return;
+  }
+
+  console.log(new Error('foo'));
```

Comment:
Did you intend to throw for both parent and child process?

Reply:
No, only on the child process. And no throwing is required, since we only like to check on the stack traces. So here we just log the error out and exit properly.